selectors/posts.ts

```diff
@@ -46,3 +48,23 @@ export function isInlineImageVisible(state: GlobalState, postId: string, imageKe
 
     return getGlobalItem(state, StoragePrefixes.INLINE_IMAGE_VISIBLE + currentUserId + '_' + postId + '_' + imageKey, !imageCollapsed);
 }
+
+export const showPostPriorityTooltip = createSelector(
+    'showPostPriorityTooltip',
+    (state: GlobalState) => get(state, Preferences.CATEGORY_POST_PRIORITY, Preferences.NAME_POST_PRIORITY_TUTORIAL_STATE, false),
+    getCurrentUser,
+    (tutorialState, user) => {
+        if (tutorialState && JSON.parse(tutorialState)[Preferences.POST_PRIORITY_VIEWED]) {
+            return false;
+        }
+        const createAt = new Date(user.create_at).getTime();
+        const now = new Date().getTime();
```

Comment:
0/5 - There's not really a better way to do this, but I just wanted to point out that this will only get recalculated when either the current user or the value of that preference does. Most of the time, this mostly means that the user won't see it until they first load the page after that week threshold has passed.

I think that's probably fine though given that this doesn't seem like a time sensitive thing we need to show someone immediately, but I'll leave that if it's up to you if it's worth changing. If you do want that to work without a refresh though, you should add a third input selector and have the boolean value calculated below. That way, it'll still only recalculate the rest of this (and re-parse the JSON) once when that first week has passed. If we had the selector, for example, return the difference between the times, then it would recalculate constantly, so we may as well not use `createSelector` at that point